### PR TITLE
Cal Update Service Account

### PR DESCRIPTION
**Pull request by scottilee**
Updates the code to use a service account.

@jlewi This script would need to be passed a service account file created on the Google account that hosts the Kubeflow calendar and the calendar would need to be shared with the service account email address. Is this something you could help me with?

/assign @jlewi

Related to #274.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/community/288)

### CODE REVIEW

**Comment by jlewi** [comment]
@scottilee @theadactyl owns the calendar. Can we use  a GitHub issue to track what actions are needed? 

It looks like we need to figure out. We might want to setup a new Kubeflow community project to host this infra. Lets use the issue to track that and figure it out.

File: scripts/calendar_import.py [hunk 1]
```diff
@@ -4,20 +4,26 @@
...
 from datetime import datetime
 import logging
 import os.path
 import yaml
 import googleapiclient.errors
 from googleapiclient.discovery import build
-from google_auth_oauthlib.flow import InstalledAppFlow
+from oauth2client.service_account import ServiceAccountCredentials
 
+SERVICE_ACCOUNT_FILE = 'kubeflow_calendar_sa.json'
```
Comment (jlewi) [inline scripts/calendar_import.py]: Why are you hard coding the service_account file.

The convention is to look for a GOOGLE_APPLICATION_CREDENTIALS environment variable which is pointing at the key.

Moving forward though we'd actually like to use GKE Workload identity;
https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity

Do you know if this code will work with Workload Identity?

Comment (scottilee) [reply scripts/calendar_import.py]: I read over that page and how I understand it is that a service account is available to a pod... so I'm assuming the service account could still be referred to as `GOOGLE_APPLICATION_CREDENTIALS`? If so, that should work.

Comment (jlewi) [reply scripts/calendar_import.py]: When workload identity is used the identity will not be determined by GOOGLE_APPLICATION_CREDENTIALS

Instead the pod should have a KSA (Kubernetes Service Account) that is mapped to a GSA(Google Service Account).  Google client libraries should be initiated to use the application default credentials in which case the credential should map to whatever GSA is mapped to the KSA.

At any rate I think we can figure it out when we get there.

**Comment by jlewi** [comment]
@scottilee What is the next step with this PR?

**Comment by jlewi** [comment]
I pulled these changes into kubeflow/community#298 which includes additional changes so I think we can go ahead and close this PR.

**Pull request closed without merging**